src/MagnumPlugins/DdsImporter/DdsImporter.cpp

```diff
@@ -0,0 +1,351 @@
+    _volume = ((ddsh.caps2 & DdsCaps2::Volume) && (ddsh.depth > 0));
+
+    /* check if image is a cubemap */
+    const bool isCubemap = (ddsh.caps2 & DdsCaps2::Cubemap);
+
+    /* set the color format */
+    _components = 4;
+    _compressed = false;
+    if (ddsh.ddspf.flags & DdsPixelFormat::FourCC) {
+        switch (DdsCompressionTypes(ddsh.ddspf.fourCC)) {
+            case DdsCompressionTypes::DXT1:
+                _colorFormat.compressed = CompressedColorFormat::RGBAS3tcDxt1;
+                break;
+            case DdsCompressionTypes::DXT3:
+                _colorFormat.compressed = CompressedColorFormat::RGBAS3tcDxt3;
+                break;
+            case DdsCompressionTypes::DXT5:
+                _colorFormat.compressed = CompressedColorFormat::RGBAS3tcDxt5;
+                break;
+            default:
+                Error() << "unknown texture compression '" + fourcc(ddsh.ddspf.fourCC) + "'";
+        }
+        _compressed = true;
+    } else if (ddsh.ddspf.rgbBitCount == 32 &&
+               ddsh.ddspf.rBitMask == 0x00FF0000 &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x000000FF &&
+               ddsh.ddspf.aBitMask == 0xFF000000) {
+        _colorFormat.uncompressed = ColorFormat::BGRA;
+    } else if (ddsh.ddspf.rgbBitCount == 32 &&
+               ddsh.ddspf.rBitMask == 0x000000FF &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x00FF0000 &&
+               ddsh.ddspf.aBitMask == 0xFF000000) {
+        _colorFormat.uncompressed = ColorFormat::RGBA;
+    } else if (ddsh.ddspf.rgbBitCount == 24 &&
+               ddsh.ddspf.rBitMask == 0x000000FF &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x00FF0000) {
+        _colorFormat.uncompressed = ColorFormat::RGB;
+        _components = 3;
+    } else if (ddsh.ddspf.rgbBitCount == 24 &&
+               ddsh.ddspf.rBitMask == 0x00FF0000 &&
+               ddsh.ddspf.gBitMask == 0x0000FF00 &&
+               ddsh.ddspf.bBitMask == 0x000000FF) {
+        _colorFormat.uncompressed = ColorFormat::BGR;
+        _components = 3;
+    } else if (ddsh.ddspf.rgbBitCount == 8) {
+        #ifndef MAGNUM_TARGET_GLES2
+        _colorFormat.uncompressed = ColorFormat::Red;
+        #else
+        _colorFormat.uncompressed = ColorFormat::Luminance;
+        #endif
+    } else {
+        Error() << "Unknow texture format";
+    }
+
+    const Vector3i size{Int(ddsh.width), Int(ddsh.height), Int(Math::max(ddsh.depth, 1u))};
+
+    /* check how many mipmaps to load */
+    const UnsignedInt numMipmaps = (ddsh.flags & DdsDescriptionFlag::MipMapCount)
+                    ? ((ddsh.mipMapCount == 0) ? 0 : ddsh.mipMapCount - 1)
+                    : 0;
+
+    /* load all surfaces for the image (6 surfaces for cubemaps) */
+    const UnsignedInt numImages = (isCubemap) ? 6 : 1;
+    size_t offset = 128;
```

Comment:
Can the `128` be represented using some calculation? E.g. `sizeof(DdsHeader) + something`? Ideally in combination with `static_assert()` that it it really equals to 128.